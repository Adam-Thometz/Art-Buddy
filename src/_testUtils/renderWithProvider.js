import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;