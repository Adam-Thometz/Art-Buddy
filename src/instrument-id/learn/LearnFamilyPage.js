import React from "react";

import './LearnFamilyPage.css'

import Clickable from "../../_components/clickable-icon/Clickable";

import learnOptions from "./learnOptions";

const LearnFamilyPage = ({ family }) => {
  // debugger;
  const instrumentFamily = learnOptions[family];
  const { instruments } = instrumentFamily;
  return (
    <div className="LearnFamilyPage">
      <div className="LearnFamilyPage-header">
        <Clickable
          icon={instrumentFamily.main}
          text={`THE ${family.toUpperCase()} FAMILY`}
        />
      </div>
      <div className="LearnFamilyPage-options">
        {instruments.map(instrument => (
          <Clickable
            icon={instrument.icon}
            size='150px'
            text={instrument.name.toUpperCase()}
          />
        ))}
      </div>
    </div>
  );
};

export default LearnFamilyPage;