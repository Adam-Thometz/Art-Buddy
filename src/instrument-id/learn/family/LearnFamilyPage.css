.LearnFamilyPage {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 80%;
}

.LearnFamilyPage-header {
  width: 40%;
  font-size: 3rem;
}

.LearnFamilyPage-options {
  border-left: 1px solid #4D4F5E;
  width: 60%;
  display: grid;
  grid-template-columns: repeat(3, 33%);
  grid-template-rows: repeat(2, 50%);
}