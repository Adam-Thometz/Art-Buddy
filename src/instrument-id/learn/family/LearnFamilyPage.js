import React from "react";
import { useParams, useNavigate } from "react-router-dom";

import './LearnFamilyPage.css';

import Icon from "_components/icon/Icon";
import WindowNavbar from "_components/window-nav/WindowNavbar";
import Help from "_components/help/Help";

import learnInstrumentOptions from "../../learnInstrumentOptions";
import { instrumentIdUrls } from "_routes/routeUrls";

const LearnFamilyPage = () => {
  const navigate = useNavigate();
  const { family } = useParams();
  const instrumentFamily = learnInstrumentOptions[family];
  const { instruments } = instrumentFamily;

  const options = instruments.map(instrument => {
    const goToInstrument = () => {
      const url = `${
        instrumentIdUrls.learnUrl
      }/${
        family
      }/${
        instrument.name.toLowerCase().replace(' ', '-')
      }`;
      navigate(url);
    };

    return (<Icon
      key={instrument.name}
      icon={instrument.icon}
      size='200px'
      width={instrument.name === 'TROMBONE' ? '250px' : null}
      text={instrument.name.toUpperCase()}
      onClick={goToInstrument}
    />);
  });
  
  return (
    <main className="LearnFamilyPage">
      <WindowNavbar page='INSTRUMENT ID: LEARN' cornerIcon={<Help />} />
      <div className="LearnFamilyPage-main">
        <header className="LearnFamilyPage-header">
          <Icon icon={instrumentFamily.main} text={`THE ${family.toUpperCase()} FAMILY`} largeFont />
        </header>
        <section className="LearnFamilyPage-options">
          {options}
        </section>
      </div>
    </main>
  );
};

export default LearnFamilyPage;