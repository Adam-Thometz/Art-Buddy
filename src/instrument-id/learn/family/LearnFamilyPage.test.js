import React from "react";

import renderWithProvider from "../../../_testUtils/renderWithProvider";

import LearnFamilyPage from "./LearnFamilyPage";

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    family: 'brass'
  })
}));

describe('LearnFamilyPage component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<LearnFamilyPage />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<LearnFamilyPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});