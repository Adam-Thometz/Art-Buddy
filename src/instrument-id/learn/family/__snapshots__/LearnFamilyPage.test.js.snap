// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnFamilyPage component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LearnFamilyPage"
  >
    <div
      class="LearnFamilyPage-header"
    >
      <div
        class="Icon"
      >
        <img
          alt=""
          src="[object Object]"
          style="width: 250px; height: 250px;"
        />
        <p>
          THE BRASS FAMILY
        </p>
      </div>
    </div>
    <div
      class="LearnFamilyPage-options"
    >
      <div
        class="Icon"
      >
        <img
          alt=""
          src="[object Object]"
          style="width: 150px; height: 150px; cursor: pointer;"
        />
        <p>
          TRUMPET
        </p>
      </div>
      <div
        class="Icon"
      >
        <img
          alt=""
          src="[object Object]"
          style="width: 150px; height: 150px; cursor: pointer;"
        />
        <p>
          FRENCH HORN
        </p>
      </div>
      <div
        class="Icon"
      >
        <img
          alt=""
          src="[object Object]"
          style="width: 150px; height: 150px; cursor: pointer;"
        />
        <p>
          TROMBONE
        </p>
      </div>
      <div
        class="Icon"
      >
        <img
          alt=""
          src="[object Object]"
          style="width: 150px; height: 150px; cursor: pointer;"
        />
        <p>
          TUBA
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
